plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/AwsEc2ServiceImpl.java

```diff
@@ -57,15 +60,36 @@ public synchronized AmazonEC2 client() {
             return client;
         }
 
-        this.client = new AmazonEC2Client(buildCredentials(logger, settings), buildConfiguration(logger, settings));
+        AmazonEC2ClientBuilder builder = AmazonEC2ClientBuilder.standard()
+            .withCredentials(buildCredentials(logger, settings))
+            .withClientConfiguration(buildConfiguration(logger, settings));
+
         String endpoint = findEndpoint(logger, settings);
         if (endpoint != null) {
-            client.setEndpoint(endpoint);
+            String region = buildRegion(endpoint);
+            // TODO according to the documentation we should set the region instead of setting a endpoint
```

Comment:
Part of the reason we moved away from region was it required bumping the aws client dep whenever a new region was added. By setting endpoint, we avoid that.  For this reason, I do not think we should go back to setting region.

Reply:
Make sense.